app/models/miq_report/formats.rb

```diff
@@ -1,19 +1,27 @@
 class MiqReport::Formats
-  format_hash = YAML.load_file(ApplicationRecord::FIXTURE_DIR.join('miq_report_formats.yml')).freeze
-  FORMATS                = format_hash[:formats].freeze
-  DEFAULTS_AND_OVERRIDES = format_hash[:defaults_and_overrides].freeze
+  private_class_method def self.format_hash
+    @format_hash ||= YAML.load_file(ApplicationRecord::FIXTURE_DIR.join('miq_report_formats.yml')).freeze
```

Comment:
ooh, delay load... are you sure no one was calling FORMATS or DEFAULTS_AND_OVERRIDES from outside of this file?

I don't see any but this looks too good to be true.

Reply:
yeah i searched the entire code base for FORMATS and DEFAULTS_AND_OVERRIDES and it was all in this file.

Reply:
https://github.com/search?q=org%3AManageIQ+FORMATS&type=code (this one takes a while to go through)

https://github.com/search?q=org%3AManageIQ+DEFAULTS_AND_OVERRIDES&type=code